projectmainpage.html

```diff
@@ -0,0 +1,265 @@
+                    <li><input type="checkbox" id="20-50" name="20-50"> $20 - $50</li>
+                    <li><input type="checkbox" id="50-plus" name="50-plus"> $50 and above</li>
+                </ul>
+            </aside>
+
+            <!-- recommendations section with headers-->
+            <section id="home-recommendations">
+                <h2>Welcome to Makeup Manager!</h2>
+                <br>
+                <br>
+                <h2>Makeup reccomendations for you: </h2>
+                <br>
+                <br>
+                <div id="recommendation-grid"></div>
+            </section>
+        </main>
+
+        <br>
+        <!-- Footer -->
+        <footer id="footer">
+            <p>Copyright by MakeupManager</p>
+        </footer>
+
+        <script>
+            const recommendationsGrid = document.getElementById('recommendation-grid');
+            const filterContainer = document.getElementById('filter-container');
+            const searchBar = document.getElementById('search-bar');
+            const loadingPopup = document.getElementById('loading-popup');
+
+            let allProducts = [];
+            let filteredProducts = [];
+            let savedProducts = []; 
+
+            function showLoading() {
+                loadingPopup.style.display = 'flex';
+                loadingPopup.innerText = 'Loading recommendations, please wait...';
+            }
+
+            function hideLoading() {
+                loadingPopup.style.display = 'none';
+            }
+
+            showLoading();
+            fetch('http://makeup-api.herokuapp.com/api/v1/products.json')
+                .then(response => response.json())
+                .then(data => {
+                    allProducts = data.map(product => {
+                        if (product.product_type === "foundation") {
+                            if (product.name.toLowerCase().includes("concealer")) {
+                                product.category = "concealer";
+                            } else if (product.name.toLowerCase().includes("bronzer")) {
+                                product.category = "bronzer";
+                            } else {
+                                product.category = "foundation";
+                            }
+                        } else {
+                            product.category = product.product_type;
+                        }
+                        return product;
+                    });
+
+                    filteredProducts = allProducts;
+                    displayProducts(filteredProducts);
+                    hideLoading();
+                })
+                .catch(error => {
+                    console.error('Error fetching data:', error);
+                    loadingPopup.innerText = 'Failed to load recommendations. Please try again later.';
+                });
+
+            function displayProducts(products) {
```

Comment:
This is blocking code. We want our users to be able to use the website as soon as possible, so this should be executed asynchronously, think of this like a "dispatch" task.

Reply:
fixed